lalrpop/src/grammar/parse_tree.rs

```diff
@@ -253,6 +310,15 @@ pub enum TerminalString {
     Error,
 }
 
+impl TerminalString {
+    pub fn as_literal(&self) -> Option<&TerminalLiteral> {
```

Comment:
Just FYI, `TerminalLiteral` is `Copy`, so you don't need to return an `Option<&TerminalLiteral>` -- just `Option<TerminalLiteral>` would suffice (and then no `ref` on the pattern below)